### DISCUSSION TITLE
[SOLVED] Is there a way to set Min and Max for TimeSpan and/or TimeEdit?

### DISCUSSION THREAD

**Discussion by gac-noob** [Megabit/Blazorise#2020] [Q&A] [answered]
Hi,

Is there a way to set **Min** and **Max** value the user can set in a `<TimeEdit>` control for a `TimeSpan?` property?

Example: If I have two (2) `TimeSpan?` properties...
```
public TimeSpan? timeStart { get; set; }
public TimeSpan? timeEnd { get; set; }
```

And in the case for example the user sets the `<TimeEdit>` input control for `timeStart` to **8:00AM**...
I'd like to prevent the user from setting the `<TimeEdit>` input control for `timeEnd` to 8:00AM or earlier, the user must only be able to set it to 8:01AM (the min because timeStart is 8:00AM) to 5:00PM (the max is preset) only

Same also if the user sets the `<TimeEdit>` input control for `timeEnd` to **11:00AM**...
I'd like to prevent the user from setting the `<TimeEdit>` input control for `timeStart` to 11:00AM or later, the user must only be able to set it to 8:00AM (the min is preset) to 10:59AM (the max because timeEnd is 11:00AM) only

**Comment by stsrki** [comment, marked as answer]
Currently, it is not possible. I'm working on this feature at the moment and it will be available in the next milestone.

**Reply by gac-noob** [reply]
Yahooo!! Looking forward to it...thank you very much sir @stsrki !

By the way sir mladen, I apologize for asking it here, because this is not related to blazorize, but I'll just ask it here anyway because maybe you can advise or teach me :) ...I'm using this timeStart and timeEnd inside a class called RaceEvent...

```
public class RaceEvent
{
    public RaceEvent()
    {
        EventName = "";
        TimeStart= null;
        TimeEnd = null;
    }

    public string EventName { get; set; }
    public TimeSpan? TimeStart { get; set; }
    public TimeSpan? TimeEnd { get; set; }

    /* This is my Computed Property */
    public double EventDuration
    {
        get
        {
            if (TimeStart != null && TimeEnd != null)
            {
                var result = TimeEnd.Value.Subtract(TimeStart.Value); 
                return result.TotalMinutes <= 0 ? 0 : result.TotalMinutes; 
            }

            return 0;
        }
    }
}
```

I am using blazor client-side and passing data to SQL server thru an ASP.NET Core web api controller..and everything works perfectly...my razor page contains two (2) TimeEdit for TimeStart and TimeEnd, then I have a NumericEdit for EventDuration (display purposes only)

The difference is:

1. When I am editing a record (meaning the razor page has retrieved a record from the sql database thur the webapi) changing values in the TimeEdit automatically updates the NumericEdit EventDuration in real-time..it automatically calculates and updates the value of EventDuration as I change the values of TimeStart and TimeEnd...this is perfect
2. But when I create a NEW record (meaning the razor page has not yet posted the RaceEvent object to the webapi and the sql database) changing the values of TimeStart and TimeEnd does NOT automatically update the computed property EventDuration..so when I am changing the values of TimeStart and TimeEnd, EventDuration is always zero (0)..if I save the record, and proceed to open and edit the record, the computed property EventDuration will work... :(

I'm 100% sure this is not related to Blazorise..any idea why this is happening? 
Thanks in advance sir mladen!

**Reply by gac-noob** [reply]
Sorry2 sir @stsrki ...I found the problem..the cause is because I'm an IDIOT! I messed up my create.razor page! hahaha thanks again!!
